apps/dashboard/src/components/workflow-list.tsx

```diff
@@ -158,7 +160,18 @@ export const WorkflowList = () => {
                             <FaCode className="size-3" />
                           </Badge>
                         )}
-                        <TruncatedText text={workflow.name} />
+                        <TruncatedText
+                          className="cursor-pointer"
+                          text={workflow.name}
+                          onClick={() => {
+                            navigate(
+                              buildRoute(ROUTES.EDIT_WORKFLOW, {
+                                environmentId: currentEnvironment?._id ?? '',
```

Comment:
@SokratisVidros, it's not that simple:
1. Throwing an error won't work, because on the initial render the `currentEnvironment` is always undefined, so the `useEnvironment` hook will always throw.
2. That's a partial solution to this problem because all code above the `if (!isLoaded)` including hooks will still need to check whether the current env is present.

Reply:
We will take this offline. It's not a good enough blocker for this PR.